### PR TITLE
Add getters for AuthorizationStatus and AuthorizationError Structs

### PR DESCRIPTION
**Pull request by rahulanand16nov**
After parsing an 'authorize' response from 3scale, there was no way to access fields inside the `OkAuthorization` struct. In this PR, I have added getters to give access to these fields as references. Please let me know if there is a change required.

### CODE REVIEW

File: src/response.rs
Comment (davidor) [inline src/response.rs:118]: Shouldn't these 2 be `pub` ?

File: src/response.rs
Comment (davidor) [inline src/response.rs:122]: Typo in function name

Comment (rahulanand16nov) [reply src/response.rs]: 😬Sorry about that! Fixed

**Comment by unleashed** [comment]
There's quite a few issues with the current `Authorization` response parsing - please wait until #90 is merged and rebase on top of it (and quite possibly keep adding more getters, since the refactoring and support added there don't address this) because this PR is way smaller and easier to rebase.

**Comment by unleashed** [comment]
#90 has been merged and now this PR should be rebased on top of the latest master.

**Comment by rahulanand16nov** [comment]
Rebased on top of remote master and squashed.

**Review by unleashed — COMMENTED** [review]
What about other private fields like the boolean indicating whether the request has been authorized and the list of app keys, etc?

**Comment by rahulanand16nov** [comment]
@unleashed Added.

**Review by unleashed — APPROVED** [review]

**Comment by unleashed** [comment]
@davidor ping for review?

**Review by davidor — APPROVED** [review]

**Comment by davidor** [comment]
bors r=@unleashed,@davidor

**Pull request merged**
